<Type Name="NotFiniteNumberException" FullName="System.NotFiniteNumberException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="740346ece36fb0de37bf1ea74a2d2e9f7cb9a7c2" /><Meta Name="ms.sourcegitcommit" Value="bac13a81377b1c4ebf3990e3287a6f1700f73751" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="ru-RU" /><Meta Name="ms.lasthandoff" Value="03/19/2019" /><Meta Name="ms.locfileid" Value="57929846" /></Metadata><TypeSignature Language="C#" Value="public class NotFiniteNumberException : ArithmeticException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit NotFiniteNumberException extends System.ArithmeticException" />
  <TypeSignature Language="DocId" Value="T:System.NotFiniteNumberException" />
  <TypeSignature Language="VB.NET" Value="Public Class NotFiniteNumberException&#xA;Inherits ArithmeticException" />
  <TypeSignature Language="C++ CLI" Value="public ref class NotFiniteNumberException : ArithmeticException" />
  <TypeSignature Language="F#" Value="type NotFiniteNumberException = class&#xA;    inherit ArithmeticException" />
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.ArithmeticException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netframework-4.8">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
    <Attribute FrameworkAlternate="netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0">
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="c3119-101">Исключение, которое выбрасывается, когда значение с плавающей запятой является плюс бесконечностью, минус бесконечностью или не является числовым (NaN).</span><span class="sxs-lookup"><span data-stu-id="c3119-101">The exception that is thrown when a floating-point value is positive infinity, negative infinity, or Not-a-Number (NaN).</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="c3119-102"><xref:System.NotFiniteNumberException> доступен для языков программирования, которые не поддерживают концепции бесконечности и не является числовым в операции с плавающей запятой.</span><span class="sxs-lookup"><span data-stu-id="c3119-102"><xref:System.NotFiniteNumberException> is available for programming languages that do not support the concepts of infinity and Not-a-Number in floating-point operations.</span></span>  
  
 <span data-ttu-id="c3119-103"><xref:System.NotFiniteNumberException> использует COR_E_NOTFINITENUMBER HRESULT, которое имеет значение 0x80131528.</span><span class="sxs-lookup"><span data-stu-id="c3119-103"><xref:System.NotFiniteNumberException> uses the HRESULT COR_E_NOTFINITENUMBER, which has the value 0x80131528.</span></span>  
  
 <span data-ttu-id="c3119-104">Список начальных значений свойств для экземпляра <xref:System.NotFiniteNumberException>, см. в разделе <xref:System.NotFiniteNumberException.%23ctor%2A> конструкторы.</span><span class="sxs-lookup"><span data-stu-id="c3119-104">For a list of initial property values for an instance of <xref:System.NotFiniteNumberException>, see the <xref:System.NotFiniteNumberException.%23ctor%2A> constructors.</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
    <altmember cref="T:System.Double" />
    <altmember cref="T:System.Single" />
    <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="c3119-105">Обработка и создание исключений в .NET</span><span class="sxs-lookup"><span data-stu-id="c3119-105">Handling and throwing exceptions in .NET</span></span></related>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="c3119-106">Инициализирует новый экземпляр класса <see cref="T:System.NotFiniteNumberException" />.</span><span class="sxs-lookup"><span data-stu-id="c3119-106">Initializes a new instance of the <see cref="T:System.NotFiniteNumberException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public NotFiniteNumberException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NotFiniteNumberException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; NotFiniteNumberException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="c3119-107">Инициализирует новый экземпляр класса <see cref="T:System.NotFiniteNumberException" />.</span><span class="sxs-lookup"><span data-stu-id="c3119-107">Initializes a new instance of the <see cref="T:System.NotFiniteNumberException" /> class.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="c3119-108">Этот конструктор инициализирует <xref:System.Exception.Message%2A> свойства нового экземпляра системное сообщение с описанием ошибки, например, «возникшее число не было конечной величиной».</span><span class="sxs-lookup"><span data-stu-id="c3119-108">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "The number encountered was not a finite quantity."</span></span> <span data-ttu-id="c3119-109">Это сообщение учитывает текущую культуру системы.</span><span class="sxs-lookup"><span data-stu-id="c3119-109">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="c3119-110">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.NotFiniteNumberException>.</span><span class="sxs-lookup"><span data-stu-id="c3119-110">The following table shows the initial property values for an instance of <xref:System.NotFiniteNumberException>.</span></span>  
  
|<span data-ttu-id="c3119-111">Свойство.</span><span class="sxs-lookup"><span data-stu-id="c3119-111">Property</span></span>|<span data-ttu-id="c3119-112">Значение</span><span class="sxs-lookup"><span data-stu-id="c3119-112">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="c3119-113">Пустая ссылка (`Nothing` в Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="c3119-113">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="c3119-114">Строка локализованного сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="c3119-114">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public NotFiniteNumberException (double offendingNumber);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(float64 offendingNumber) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NotFiniteNumberException.#ctor(System.Double)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (offendingNumber As Double)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; NotFiniteNumberException(double offendingNumber);" />
      <MemberSignature Language="F#" Value="new NotFiniteNumberException : double -&gt; NotFiniteNumberException" Usage="new System.NotFiniteNumberException offendingNumber" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="offendingNumber" Type="System.Double" />
      </Parameters>
      <Docs>
        <param name="offendingNumber"><span data-ttu-id="c3119-115">Значение аргумента, вызвавшего исключение.</span><span class="sxs-lookup"><span data-stu-id="c3119-115">The value of the argument that caused the exception.</span></span></param>
        <summary><span data-ttu-id="c3119-116">Инициализирует новый экземпляр класса <see cref="T:System.NotFiniteNumberException" /> недопустимым числом.</span><span class="sxs-lookup"><span data-stu-id="c3119-116">Initializes a new instance of the <see cref="T:System.NotFiniteNumberException" /> class with the invalid number.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="c3119-117">`offendingNumber` Должен быть <xref:System.Double> и недопустимым числом.</span><span class="sxs-lookup"><span data-stu-id="c3119-117">The `offendingNumber` parameter must be both a <xref:System.Double> and an invalid number.</span></span>  
  
 <span data-ttu-id="c3119-118">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.NotFiniteNumberException>.</span><span class="sxs-lookup"><span data-stu-id="c3119-118">The following table shows the initial property values for an instance of <xref:System.NotFiniteNumberException>.</span></span>  
  
|<span data-ttu-id="c3119-119">Свойство.</span><span class="sxs-lookup"><span data-stu-id="c3119-119">Property</span></span>|<span data-ttu-id="c3119-120">Значение</span><span class="sxs-lookup"><span data-stu-id="c3119-120">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="c3119-121">Пустая ссылка (`Nothing` в Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="c3119-121">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.NotFiniteNumberException.OffendingNumber%2A>|<span data-ttu-id="c3119-122">Недопустимое число.</span><span class="sxs-lookup"><span data-stu-id="c3119-122">The invalid number.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public NotFiniteNumberException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NotFiniteNumberException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; NotFiniteNumberException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new NotFiniteNumberException : string -&gt; NotFiniteNumberException" Usage="new System.NotFiniteNumberException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="c3119-123">Сообщение, описывающее ошибку.</span><span class="sxs-lookup"><span data-stu-id="c3119-123">The message that describes the error.</span></span></param>
        <summary><span data-ttu-id="c3119-124">Инициализирует новый экземпляр класса <see cref="T:System.NotFiniteNumberException" /> с указанным сообщением об ошибке.</span><span class="sxs-lookup"><span data-stu-id="c3119-124">Initializes a new instance of the <see cref="T:System.NotFiniteNumberException" /> class with a specified error message.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="c3119-125">Убедитесь, что `message` строка понятно пользователю, и его локализована для текущего языка и региональных параметров.</span><span class="sxs-lookup"><span data-stu-id="c3119-125">Ensure that the `message` string is understandable to the user and that it is localized for the current culture.</span></span>  
  
 <span data-ttu-id="c3119-126">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.NotFiniteNumberException>.</span><span class="sxs-lookup"><span data-stu-id="c3119-126">The following table shows the initial property values for an instance of <xref:System.NotFiniteNumberException>.</span></span>  
  
|<span data-ttu-id="c3119-127">Свойство.</span><span class="sxs-lookup"><span data-stu-id="c3119-127">Property</span></span>|<span data-ttu-id="c3119-128">Значение</span><span class="sxs-lookup"><span data-stu-id="c3119-128">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="c3119-129">Пустая ссылка (`Nothing` в Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="c3119-129">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="c3119-130">Строка сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="c3119-130">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected NotFiniteNumberException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NotFiniteNumberException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; NotFiniteNumberException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new NotFiniteNumberException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; NotFiniteNumberException" Usage="new System.NotFiniteNumberException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Security.SecuritySafeCritical</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="c3119-131">Объект, содержащий сериализованные данные объекта.</span><span class="sxs-lookup"><span data-stu-id="c3119-131">The object that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="c3119-132">Контекстные сведения об источнике или назначении.</span><span class="sxs-lookup"><span data-stu-id="c3119-132">The contextual information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="c3119-133">Инициализирует новый экземпляр класса <see cref="T:System.NotFiniteNumberException" /> с сериализованными данными.</span><span class="sxs-lookup"><span data-stu-id="c3119-133">Initializes a new instance of the <see cref="T:System.NotFiniteNumberException" /> class with serialized data.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="c3119-134">Данный конструктор вызывается в процессе десериализации для воссоздания объекта исключения, переданного в потоке.</span><span class="sxs-lookup"><span data-stu-id="c3119-134">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public NotFiniteNumberException (string message, double offendingNumber);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, float64 offendingNumber) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NotFiniteNumberException.#ctor(System.String,System.Double)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, offendingNumber As Double)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; NotFiniteNumberException(System::String ^ message, double offendingNumber);" />
      <MemberSignature Language="F#" Value="new NotFiniteNumberException : string * double -&gt; NotFiniteNumberException" Usage="new System.NotFiniteNumberException (message, offendingNumber)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="offendingNumber" Type="System.Double" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="c3119-135">Сообщение, описывающее ошибку.</span><span class="sxs-lookup"><span data-stu-id="c3119-135">The message that describes the error.</span></span></param>
        <param name="offendingNumber"><span data-ttu-id="c3119-136">Значение аргумента, вызвавшего исключение.</span><span class="sxs-lookup"><span data-stu-id="c3119-136">The value of the argument that caused the exception.</span></span></param>
        <summary><span data-ttu-id="c3119-137">Инициализирует новый экземпляр класса <see cref="T:System.NotFiniteNumberException" /> указанным сообщением об ошибке и недопустимым числом.</span><span class="sxs-lookup"><span data-stu-id="c3119-137">Initializes a new instance of the <see cref="T:System.NotFiniteNumberException" /> class with a specified error message and the invalid number.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="c3119-138">Убедитесь, что `message` строка понятно пользователю, и его локализована для текущего языка и региональных параметров.</span><span class="sxs-lookup"><span data-stu-id="c3119-138">Ensure that the `message` string is understandable to the user and that it is localized for the current culture.</span></span>  
  
 <span data-ttu-id="c3119-139">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.NotFiniteNumberException>.</span><span class="sxs-lookup"><span data-stu-id="c3119-139">The following table shows the initial property values for an instance of <xref:System.NotFiniteNumberException>.</span></span>  
  
|<span data-ttu-id="c3119-140">Свойство.</span><span class="sxs-lookup"><span data-stu-id="c3119-140">Property</span></span>|<span data-ttu-id="c3119-141">Значение</span><span class="sxs-lookup"><span data-stu-id="c3119-141">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="c3119-142">Пустая ссылка (`Nothing` в Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="c3119-142">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="c3119-143">Строка сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="c3119-143">The error message string.</span></span>|  
|<xref:System.NotFiniteNumberException.OffendingNumber%2A>|<span data-ttu-id="c3119-144">Недопустимое число.</span><span class="sxs-lookup"><span data-stu-id="c3119-144">The invalid number.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public NotFiniteNumberException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NotFiniteNumberException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; NotFiniteNumberException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new NotFiniteNumberException : string * Exception -&gt; NotFiniteNumberException" Usage="new System.NotFiniteNumberException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" Index="0" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;netstandard-2.0;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0" />
        <Parameter Name="innerException" Type="System.Exception" Index="1" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;netstandard-2.0;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="c3119-145">Сообщение об ошибке, указывающее причину создания исключения.</span><span class="sxs-lookup"><span data-stu-id="c3119-145">The error message that explains the reason for the exception.</span></span></param>
        <param name="innerException"><span data-ttu-id="c3119-146">Исключение, которое является причиной текущего исключения.</span><span class="sxs-lookup"><span data-stu-id="c3119-146">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="c3119-147">Если параметр <paramref name="innerException" /> не является указателем null (<see langword="Nothing" /> в Visual Basic), то текущее исключение создается в блоке <see langword="catch" />, обрабатывающем внутреннее исключение.</span><span class="sxs-lookup"><span data-stu-id="c3119-147">If the <paramref name="innerException" /> parameter is not a null reference (<see langword="Nothing" /> in Visual Basic), the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="c3119-148">Инициализирует новый экземпляр класса <see cref="T:System.NotFiniteNumberException" />, используя указанное сообщение об ошибке и ссылку на внутреннее исключение, которое стало исходной причиной данного исключения.</span><span class="sxs-lookup"><span data-stu-id="c3119-148">Initializes a new instance of the <see cref="T:System.NotFiniteNumberException" /> class with a specified error message and a reference to the inner exception that is root cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="c3119-149">Исключение, созданное как прямой результат предыдущего исключения, может содержать в свойстве <xref:System.Exception.InnerException%2A> ссылку на предыдущее исключение.</span><span class="sxs-lookup"><span data-stu-id="c3119-149">An exception that is thrown as a direct result of a previous exception can include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="c3119-150">Свойство <xref:System.Exception.InnerException%2A> возвращает значение, переданное конструктору, или значение `null`, если свойство <xref:System.Exception.InnerException%2A> не передает конструктору значение внутреннего исключения.</span><span class="sxs-lookup"><span data-stu-id="c3119-150">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="c3119-151">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.NotFiniteNumberException>.</span><span class="sxs-lookup"><span data-stu-id="c3119-151">The following table shows the initial property values for an instance of <xref:System.NotFiniteNumberException>.</span></span>  
  
|<span data-ttu-id="c3119-152">Свойство.</span><span class="sxs-lookup"><span data-stu-id="c3119-152">Property</span></span>|<span data-ttu-id="c3119-153">Значение</span><span class="sxs-lookup"><span data-stu-id="c3119-153">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="c3119-154">Ссылка на внутреннее исключение.</span><span class="sxs-lookup"><span data-stu-id="c3119-154">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="c3119-155">Строка сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="c3119-155">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public NotFiniteNumberException (string message, double offendingNumber, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, float64 offendingNumber, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NotFiniteNumberException.#ctor(System.String,System.Double,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, offendingNumber As Double, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; NotFiniteNumberException(System::String ^ message, double offendingNumber, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new NotFiniteNumberException : string * double * Exception -&gt; NotFiniteNumberException" Usage="new System.NotFiniteNumberException (message, offendingNumber, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="offendingNumber" Type="System.Double" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="c3119-156">Сообщение об ошибке, указывающее причину создания исключения.</span><span class="sxs-lookup"><span data-stu-id="c3119-156">The error message that explains the reason for the exception.</span></span></param>
        <param name="offendingNumber"><span data-ttu-id="c3119-157">Значение аргумента, вызвавшего исключение.</span><span class="sxs-lookup"><span data-stu-id="c3119-157">The value of the argument that caused the exception.</span></span></param>
        <param name="innerException"><span data-ttu-id="c3119-158">Исключение, которое является причиной текущего исключения.</span><span class="sxs-lookup"><span data-stu-id="c3119-158">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="c3119-159">Если параметр <paramref name="innerException" /> не является указателем null (<see langword="Nothing" /> в Visual Basic), то текущее исключение создается в блоке <see langword="catch" />, обрабатывающем внутреннее исключение.</span><span class="sxs-lookup"><span data-stu-id="c3119-159">If the <paramref name="innerException" /> parameter is not a null reference (<see langword="Nothing" /> in Visual Basic), the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="c3119-160">Инициализирует новый экземпляр класса <see cref="T:System.NotFiniteNumberException" /> указанным сообщением об ошибке, недопустимым числом и ссылкой на внутреннее исключение, вызвавшее данное исключение.</span><span class="sxs-lookup"><span data-stu-id="c3119-160">Initializes a new instance of the <see cref="T:System.NotFiniteNumberException" /> class with a specified error message, the invalid number, and a reference to the inner exception that is root cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="c3119-161">Исключение, созданное как прямой результат предыдущего исключения, может содержать в свойстве <xref:System.Exception.InnerException%2A> ссылку на предыдущее исключение.</span><span class="sxs-lookup"><span data-stu-id="c3119-161">An exception that is thrown as a direct result of a previous exception can include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="c3119-162">Свойство <xref:System.Exception.InnerException%2A> возвращает значение, переданное конструктору, или значение `null`, если свойство <xref:System.Exception.InnerException%2A> не передает конструктору значение внутреннего исключения.</span><span class="sxs-lookup"><span data-stu-id="c3119-162">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="c3119-163">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.NotFiniteNumberException>.</span><span class="sxs-lookup"><span data-stu-id="c3119-163">The following table shows the initial property values for an instance of <xref:System.NotFiniteNumberException>.</span></span>  
  
|<span data-ttu-id="c3119-164">Свойство.</span><span class="sxs-lookup"><span data-stu-id="c3119-164">Property</span></span>|<span data-ttu-id="c3119-165">Значение</span><span class="sxs-lookup"><span data-stu-id="c3119-165">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="c3119-166">Ссылка на внутреннее исключение.</span><span class="sxs-lookup"><span data-stu-id="c3119-166">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="c3119-167">Строка сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="c3119-167">The error message string.</span></span>|  
|<xref:System.NotFiniteNumberException.OffendingNumber%2A>|<span data-ttu-id="c3119-168">Недопустимое число.</span><span class="sxs-lookup"><span data-stu-id="c3119-168">The invalid number.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="c3119-169">Обработка и создание исключений в .NET</span><span class="sxs-lookup"><span data-stu-id="c3119-169">Handling and throwing exceptions in .NET</span></span></related>
      </Docs>
    </Member>
    <Member MemberName="GetObjectData">
      <MemberSignature Language="C#" Value="public override void GetObjectData (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig virtual instance void GetObjectData(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.NotFiniteNumberException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Public Overrides Sub GetObjectData (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; override void GetObjectData(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="override this.GetObjectData : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; unit" Usage="notFiniteNumberException.GetObjectData (info, context)" />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netframework-4.8">
          <AttributeName>System.Security.SecurityCritical</AttributeName>
        </Attribute>
      </Attributes>
      <ReturnValue>
        <ReturnType>System.Void</ReturnType>
      </ReturnValue>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="c3119-170">Объект, содержащий сериализованные данные объекта.</span><span class="sxs-lookup"><span data-stu-id="c3119-170">The object that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="c3119-171">Контекстные сведения об источнике или назначении.</span><span class="sxs-lookup"><span data-stu-id="c3119-171">The contextual information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="c3119-172">Задает объект <see cref="T:System.Runtime.Serialization.SerializationInfo" /> с недопустимым числом и дополнительной информацией исключения.</span><span class="sxs-lookup"><span data-stu-id="c3119-172">Sets the <see cref="T:System.Runtime.Serialization.SerializationInfo" /> object with the invalid number and additional exception information.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="c3119-173">Метод <xref:System.TypeLoadException.GetObjectData%2A> загружает в объект <xref:System.Runtime.Serialization.SerializationInfo> все данные объекта исключения, предназначенные для сериализации.</span><span class="sxs-lookup"><span data-stu-id="c3119-173"><xref:System.TypeLoadException.GetObjectData%2A> sets a <xref:System.Runtime.Serialization.SerializationInfo> with all the exception object data targeted for serialization.</span></span> <span data-ttu-id="c3119-174">Во время десериализации объект исключения воссоздается из <xref:System.Runtime.Serialization.SerializationInfo> переданного в потоке.</span><span class="sxs-lookup"><span data-stu-id="c3119-174">During deserialization, the exception object is reconstituted from the <xref:System.Runtime.Serialization.SerializationInfo> transmitted over the stream.</span></span>  
  
 <span data-ttu-id="c3119-175">Дополнительные сведения см. в разделе <xref:System.Runtime.Serialization.SerializationInfo>.</span><span class="sxs-lookup"><span data-stu-id="c3119-175">For more information, see <xref:System.Runtime.Serialization.SerializationInfo>.</span></span>  
  
 ]]></format>
        </remarks>
        <exception cref="T:System.ArgumentNullException"><span data-ttu-id="c3119-176">Объект <paramref name="info" /> имеет значение <see langword="null" />.</span><span class="sxs-lookup"><span data-stu-id="c3119-176">The <paramref name="info" /> object is <see langword="null" />.</span></span></exception>
        <permission cref="T:System.Security.SecurityCriticalAttribute"><span data-ttu-id="c3119-177">Требует полного доверия для непосредственного вызывающего объекта.</span><span class="sxs-lookup"><span data-stu-id="c3119-177">requires full trust for the immediate caller.</span></span> <span data-ttu-id="c3119-178">Этот член не может использоваться частично доверенным или прозрачным кодом.</span><span class="sxs-lookup"><span data-stu-id="c3119-178">This member cannot be used by partially trusted or transparent code.</span></span></permission>
        <altmember cref="T:System.Runtime.Serialization.SerializationInfo" />
        <altmember cref="T:System.Runtime.Serialization.StreamingContext" />
      </Docs>
    </Member>
    <Member MemberName="OffendingNumber">
      <MemberSignature Language="C#" Value="public double OffendingNumber { get; }" />
      <MemberSignature Language="ILAsm" Value=".property instance float64 OffendingNumber" />
      <MemberSignature Language="DocId" Value="P:System.NotFiniteNumberException.OffendingNumber" />
      <MemberSignature Language="VB.NET" Value="Public ReadOnly Property OffendingNumber As Double" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; property double OffendingNumber { double get(); };" />
      <MemberSignature Language="F#" Value="member this.OffendingNumber : double" Usage="System.NotFiniteNumberException.OffendingNumber" />
      <MemberType>Property</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Double</ReturnType>
      </ReturnValue>
      <Docs>
        <summary><span data-ttu-id="c3119-179">Возвращает недопустимое число, которое является плюс бесконечностью, минус бесконечностью или не является числовым (NaN).</span><span class="sxs-lookup"><span data-stu-id="c3119-179">Gets the invalid number that is a positive infinity, a negative infinity, or Not-a-Number (NaN).</span></span></summary>
        <value><span data-ttu-id="c3119-180">Недопустимое число.</span><span class="sxs-lookup"><span data-stu-id="c3119-180">The invalid number.</span></span></value>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="c3119-181"><xref:System.NotFiniteNumberException.OffendingNumber%2A> Свойство возвращает то же значение, что было передано в конструктор, или ноль, если значение было задано.</span><span class="sxs-lookup"><span data-stu-id="c3119-181">The <xref:System.NotFiniteNumberException.OffendingNumber%2A> property returns the same value as was passed into the constructor, or zero if no value was supplied.</span></span> <span data-ttu-id="c3119-182">Переопределить <xref:System.NotFiniteNumberException.OffendingNumber%2A> свойство, чтобы настроить содержимое или формат недопустимое значение.</span><span class="sxs-lookup"><span data-stu-id="c3119-182">Override the <xref:System.NotFiniteNumberException.OffendingNumber%2A> property to customize the content or format of the invalid value.</span></span>  
  
 <span data-ttu-id="c3119-183">Это свойство доступно только для чтения.</span><span class="sxs-lookup"><span data-stu-id="c3119-183">This property is read-only.</span></span>  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Double" />
      </Docs>
    </Member>
  </Members>
</Type>